File: 4-strings/5-common-problems/main.cpp
// Common problems with strings in C++
#include <iostream>
#include <string>

int main() {
    // Problem 1: Uninitialized string
    std::string uninitialized; // Default constructor initializes to empty string
    std::cout << "Uninitialized string (should be empty): '" << uninitialized << "'" << std::endl;  // Output: Uninitialized string (should be empty): ''  
    return 0;
}

File: 4-strings/5-common-problems/main.java
// Common problems with strings in Java
class CommonStringProblems {
    public static void main(String[] args) {
        // Problem 1: Uninitialized string
        String uninitialized; // Declaration without initialization
        // System.out.println("Uninitialized string: " + uninitialized); // Compile-time error  
        // Correct way: Initialize the string
        uninitialized = "";
        System.out.println("Uninitialized string (should be empty): '" + uninitialized + "'");  // Output: Uninitialized string (should be empty): ''  
    }
}